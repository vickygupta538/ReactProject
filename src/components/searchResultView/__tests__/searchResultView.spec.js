import React from "react";
import renderer from "react-test-renderer";
import { testStore } from ".././../../../test/utils";

import { Provider } from "react-redux";

import SearchResultView from "../index";

const initialState = {
  characters: [
    {
      name: "Luke Skywalker",
      height: "172",
      mass: "77",
      hair_color: "blond",
      skin_color: "fair",
      eye_color: "blue",
      birth_year: "19BBY",
      gender: "male"
    }
  ],
  selectedCharacterIndex: 0
};

const setUp = () => {
  const store = testStore(initialState);
  const component = renderer.create(
    <Provider store={store}>
      <SearchResultView />
    </Provider>
  );
  return component;
};

describe("SearchResultView Component Test Cases", () => {
  let component;

  beforeEach(() => {
    component = setUp(initialState);
  });
  it("should match snapshot renderer", () => {
    const wrapper = component.toJSON();
    expect(wrapper).toMatchSnapshot();
  });

  it("Should render without errors", () => {
    const wrapper = component.root.find(
      el => el.type === "section" && el.props["data-test"] === "SearchResultViewComponent"
    );
    expect(wrapper).toBeTruthy();
  });
});
