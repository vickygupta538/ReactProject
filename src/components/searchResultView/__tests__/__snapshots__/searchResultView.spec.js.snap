// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResultView Component Test Cases should match snapshot renderer 1`] = `
<section
  className="searchResultView"
  data-test="SearchResultViewComponent"
>
  <div>
    <strong>
      Characters:
    </strong>
  </div>
  <section
    className="searchResultItemWrapper"
    data-test="SearchResultItemComponent"
    onClick={[Function]}
  >
    <button
      className="searchResultItem"
    >
      Luke Skywalker
    </button>
  </section>
</section>
`;
